### PR TITLE
JDF-286 Picketlink/Deltaspike authorization quickstart

### PR DESCRIPTION
**Pull request by sbryzak**

### CODE REVIEW

**Comment by sgilda** [comment]
@sbryzak : The first 3 commits were also in Pull 508. I'm not a Git expert, so I'm not just evaluate the new quickstart and try to figure out how to process it separately later.

**Comment by sbryzak** [comment]
Oops, I created the new quickstart in a different branch so I thought it would be a separate pull request... I will have to do a little research on this to see if it's possible to separate pull requests.

**Comment by sgilda** [comment]
Usually a new branch prevents this. I think this happened to me once before when I did a rebase to include my changes in another pull??

**Comment by sbryzak** [comment]
I think it's possibly because the first three commits were on the master branch, not on their own branch.  Next time I'll make sure that all commits are grouped into their own branches, hopefully that should keep the pull requests separate.

**Comment by pmuir** [comment]
Do a git rebase -i HEAD~5 and delete the three commits you don't want.

**Comment by sgilda** [comment]
@pmuir : I knew you would be able to resolve this problem. :-)

**Comment by sgilda** [comment]
Very nice example. You did a great job describing how it works! Other than the class name, it looks good to me.

@pmuir : Would you like to review it?

**Comment by sgilda** [comment]
@pmuir: Was your rebase comment for me or for @sbryzak ?

File: picketlink-deltaspike-authorization/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+Security binding is DeltaSpike feature that restricts who can invoke a method (under the covers, it uses interceptors).
+
+To restrict who can invoke a method, we create an annotation, called a security binding type. This quickstart has two security binding types - `@Admin` and `@Employee`.
+
+The quickstart defines an `Authorizer` class that implements the restrictions for the security binding types. The authorizer is a CDI bean which defines methods (annotated with `@Secures) which perform the authorization checks for each security binding we create.
+
+In this quickstart the `Authorizer` we delegate authentication to JAAS, but other authentication solutions could be used.
+
+Methods on the `Controller` bean have been restricted using the security binding types.
```
Comment (sgilda) [inline picketlink-deltaspike-authorization/README.md:22]: Should you also mention that the DeltaSpike SecurityInterceptor Interceptors class has to be defined in the `beans.xml` file or is that too obvious?

**Comment by sbryzak** [comment]
The readme is correct, it is an 'Authorizer' class (the same description we gave it in the DeltaSpike docs, see http://incubator.apache.org/deltaspike/security.html) and in our case the Authorizer class happens to be called CustomAuthorizer.

As for the security interceptor, it is commented in the beans.xml file itself, however I'm not sure if simply mentioning in the readme that the "security binding uses interceptors" is sufficient.  The readme was essentially a reworked version of the one from the deltaspike-authorization example, so if we're going to elaborate on the security interceptor we'll need to do it in both places.

**Comment by sbryzak** [comment]
BTW, I think the other commits should disappear from the pull request once PR 508 is merged.  If not, I can try Pete's rebase trick.

**Comment by sgilda** [comment]
I think it's fine the way it is. Thanks!

File: picketlink-authorization-idm-jpa/src/main/java/org/jboss/as/quickstarts/picketlink/authorization/idm/jpa/IDMConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        return identityConfig;
+    }
+
+    private void initConfig() {
+        IdentityConfigurationBuilder builder = new IdentityConfigurationBuilder();
+        builder
+            .stores()
+                .jpa()
+                   .identityClass(IdentityObject.class)
```
Comment (pmuir) [inline picketlink-authorization-idm-jpa/src/main/java/org/jboss/as/quickstarts/picketlink/authorization/idm/jpa/IDMConfiguration.java:51]: Some comments here would help.

File: picketlink-authorization-idm-jpa/src/main/java/org/jboss/as/quickstarts/picketlink/authorization/idm/jpa/model/CredentialObject.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import javax.persistence.TemporalType;
+
+import org.picketlink.idm.jpa.annotations.CredentialType;
+import org.picketlink.idm.jpa.annotations.CredentialValue;
+import org.picketlink.idm.jpa.annotations.EffectiveDate;
+import org.picketlink.idm.jpa.annotations.ExpiryDate;
+import org.picketlink.idm.jpa.annotations.IdentityCredential;
+import org.picketlink.idm.jpa.annotations.Parent;
+
+/**
```
Comment (pmuir) [inline picketlink-authorization-idm-jpa/src/main/java/org/jboss/as/quickstarts/picketlink/authorization/idm/jpa/model/CredentialObject.java:38]: Missing comments

**Comment by pmuir** [comment]
Looks good but needs a lot more comments in the JPA "identity" classes.

**Comment by sbryzak** [comment]
I've added some more comments to the code, please let me know if you'd like more!

**Comment by pmuir** [comment]
I've unwound the commit mess in this pull https://github.com/jboss-jdf/jboss-as-quickstart/pull/517

**Pull request closed without merging**